src/master-playlist-controller.js

```diff
@@ -1557,4 +1582,32 @@ export class MasterPlaylistController extends videojs.EventTarget {
 
     AdCueTags.updateAdCues(media, this.cueTagsTrack_, offset);
   }
+
+  /**
+   * Calculates the desired forward buffer length based on current time
+   *
+   * @return {Number} Desired forward buffer length in seconds
+   */
+  goalBufferLength() {
```

Comment:
In the test suite i decided to set `this.goalBufferLength = MasterPlaylistController.prototype.goalBufferLength.bind(this)` to eliminate the duplication. How are you with that approach?